### PR TITLE
config:  Update remote hypervisor config

### PR DESCRIPTION
**Pull request by stevenhorsman**
- Add annotation enablement for machine_type, default_memory and default_vcpus
- Remove note that says that cpu and memory settings are ignored.

Fixes: #7256

### CODE REVIEW

File: src/runtime/config/configuration-remote.toml.in [hunk 1]
```diff
@@ -38,7 +38,7 @@ remote_hypervisor_timeout = 600
 # Each member of the list is a regular expression, which is the base name
 # of the annotation, e.g. "path" for io.katacontainers.config.hypervisor.path"
 # Note: remote hypervisor is not handling any annotations
```
Comment (bpradipt) [inline src/runtime/config/configuration-remote.toml.in:40]:
```suggestion
Note: Remote hypervisor is only handling the following annotations
```

Comment (stevenhorsman) [reply src/runtime/config/configuration-remote.toml.in]: Good catch - thanks. Updated now

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @stevenhorsman

**Review by fidencio — APPROVED** [review]
lgtm, thanks @stevenhorsman!

**Comment by stevenhorsman** [comment]
Only TDX tests are failing and they aren't using the remote hypervisor, so I think we are ready to merge it

**Pull request merged**